grapple/helpers.py

```diff
@@ -64,6 +68,9 @@ def resolve_singular(self, _, info, **kwargs):
                 if not kwargs:
                     return None
 
+                if "token" in kwargs and hasattr(cls, "get_page_from_preview_token"):
```

Comment:
Should we move this logic inside `if issubclass(cls, Page):` to only call `get_page_from_preview_token` on pages?

Reply:
good shout!